cmake/everest-generate.cmake

```diff
@@ -494,7 +507,102 @@ function (ev_add_py_module MODULE_NAME)
+    endforeach()
+
+    # Close the list
+    file(APPEND "${OUTPUT_FILE_PATH}" "]\n\n")
+
+    file(APPEND "${OUTPUT_FILE_PATH}" "[workspace.dependencies]\neverestrs = { path = \"")
+    file(APPEND "${OUTPUT_FILE_PATH}" "${everest-framework_SOURCE_DIR}/everestrs/everestrs")
+    file(APPEND "${OUTPUT_FILE_PATH}" "\" }\n")
+endfunction()
+
+function (ev_add_rs_module MODULE_NAME)
+    set(EVEREST_MODULE_INSTALL_PREFIX "${CMAKE_INSTALL_LIBEXECDIR}/everest/modules")
+
+    set(MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/${MODULE_NAME}")
+    set(MODULE_BUILD_PATH "${CMAKE_CURRENT_BINARY_DIR}/${MODULE_NAME}")
+
+    if(IS_DIRECTORY ${MODULE_PATH})
+        if(NOT ${EVEREST_ENABLE_RS_SUPPORT})
+            message(STATUS "Excluding Rust module ${MODULE_NAME} because EVEREST_ENABLE_RS_SUPPORT=${EVEREST_ENABLE_RS_SUPPORT}")
+            return()
+        elseif("${MODULE_NAME}" IN_LIST EVEREST_EXCLUDE_MODULES)
+            message(STATUS "Excluding module ${MODULE_NAME}")
+            return()
+        elseif(EVEREST_INCLUDE_MODULES AND NOT ("${MODULE_NAME}" IN_LIST EVEREST_INCLUDE_MODULES))
+            message(STATUS "Excluding module ${MODULE_NAME}")
+            return()
+        else()
+            message(STATUS "Setting up Rust module ${MODULE_NAME}")
+
+            # Symlink the source directory into the build directory
+            execute_process(COMMAND rm -f "${MODULE_BUILD_PATH}")
+            execute_process(COMMAND ln -s "${MODULE_PATH}/" "${MODULE_BUILD_PATH}")
+
+            set(EVEREST_MODULE_DIR ${PROJECT_SOURCE_DIR}/modules)
+            file(RELATIVE_PATH MODULE_PARENT_DIR ${EVEREST_MODULE_DIR} "${CMAKE_CURRENT_SOURCE_DIR}/${MODULE_NAME}")
+            get_property(RUST_MODULE_LIST_GLOBAL GLOBAL PROPERTY RUST_MODULE_LIST)
+            list(APPEND RUST_MODULE_LIST_GLOBAL "${MODULE_PARENT_DIR}")
+            set_property(GLOBAL PROPERTY RUST_MODULE_LIST ${RUST_MODULE_LIST_GLOBAL})
+
+            find_program(
+                CARGO
+                cargo
+                REQUIRED
+            )
+
+            if(CMAKE_BUILD_TYPE STREQUAL "Release")
+                set(CARGO_RELEASE_FLAG "--release")
+                set(BIN_PATH "${CMAKE_BINARY_DIR}/modules/target/release/${MODULE_NAME}")
+            else()
+                set(CARGO_RELEASE_FLAG "")
+                set(BIN_PATH "${CMAKE_BINARY_DIR}/modules/target/debug/${MODULE_NAME}")
+            endif()
+
+            # Note: We are building in the directory in which the Workspace Cargo.toml resides, to make sure it is used.
+            add_custom_target(
+                ${MODULE_NAME} ALL
+                COMMAND ${CMAKE_COMMAND} -E env EVEREST_RS_FRAMEWORK_SOURCE_LOCATION="${everest-framework_SOURCE_DIR}" EVEREST_RS_FRAMEWORK_BINARY_LOCATION="${everest-framework_BINARY_DIR}" ${CARGO} build ${CARGO_RELEASE_FLAG} --bin ${MODULE_NAME}
+                WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/modules
+                DEPENDS everest::framework
+            )
```

Comment:
exactly, add this target as a dependency to the global `everest_core_build_rust_modules` target, and then do the symlinks inside here using commands